nidirect_gp/nidirect_gp.module

```diff
@@ -33,3 +33,62 @@ function nidirect_gp_theme() {
     ],
   ];
 }
+
+/**
+ * Implements hook_preprocess_node().
+ */
+function nidirect_gp_preprocess_node(&$variables) {
+  $node = &$variables['node'];
+  if ($node->bundle() != 'gp_practice' || $variables['view_mode'] != 'teaser') {
+    return;
+  }
+
+  // Rejig the content render array based on what was presented from underlying GP entities
+  // and existing GP Practice field values.
+  $gp_details = [];
+
+  // Lead GP.
+  if (!empty($node->field_gp_practice_lead->referencedEntities())) {
+    $lead_gp = $node->field_gp_practice_lead->referencedEntities()[0];
```

Comment:
You'd be better off rendering a display mode instead of creating the 'display' in code.
Same goes for the next few if statements

Reply:
Noted, yep. I need to think about this more, but essentially I have a display mode for teaser which this node type uses but the field formatter options don't quite give me what I need.

I have more time than expected (!!) for this so what I can do next is:

- View mode for GP entity to present the first and last name to reduce the swapping out of existing items in the existing content section.
- I still need to render address components selectively in the Twig template, but that's easy to do when combined with the without filter.

Reply:
Just create a pseudo field called 'full name' and use that in the display mode.

Reply:
👍